src/Frame.js

```diff
@@ -0,0 +1,51 @@
+function Frame(finalFrame) {
+  this._rolls = [];
+  this._bonus = null;
+  this._finalFrame = finalFrame;
+};
+
+Frame.prototype.roll = function(value) {
+ this._rolls.push(value);
+};
+
+Frame.prototype.calculateFrameScore = function () {
+  return this._rolls.reduce(function(score1, score2) {
+    return score1 + score2;
+  }, 0);
+};
+
+Frame.prototype.isStrike = function () {
+  return this._rolls.length === 1 && this.calculateFrameScore() === 10;
+};
+
+Frame.prototype.isSpare = function () {
+  return this._rolls.length === 2 && this.calculateFrameScore() === 10;
+};
+
+Frame.prototype.isComplete = function () {
+  return this._rolls.length === 2 || this.isStrike();
+};
+
+Frame.prototype.bonus = function(value) {
+  if (this.isStrike()) {
+  this._bonus = value.reduce(function(score1, score2) {
```

Comment:
Could `score1` and `score2` have names that more clearly explain their role in the code?